### PR TITLE
Better Bolt 11 parsing spec compliance

### PR DESCRIPTION
**Pull request by t-bast**
Add more Bolt 11 spec tests.
Ignore hashed fields with invalid length as the spec requires:

```
A reader:
    * MUST skip over unknown fields, OR an f field with unknown version, OR p, h, s or n fields that do NOT have data_lengths of 52, 52, 52 or 53, respectively.
```

See https://github.com/lightningnetwork/lightning-rfc/pull/699 and https://github.com/lightningnetwork/lightning-rfc/pull/736.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala [hunk 1]
```diff
@@ -377,41 +390,58 @@ object PaymentRequest {
...
+
+    val paymentSecretCodec: Codec[PaymentSecretTag] = discriminatorWithDefault(
+      discriminated.by(dataLengthCodec)
+        .typecase(260, dataCodecLengthDiscriminated(260, bytes32.as[PaymentSecret])),
+      bits.as[InvalidPaymentSecret].upcast[PaymentSecretTag])
+
+    val descriptionHashCodec: Codec[DescriptionHashTag] = discriminatorWithDefault(
+      discriminated.by(dataLengthCodec)
+        .typecase(260, dataCodecLengthDiscriminated(260, bytes32.as[DescriptionHash])),
+      bits.as[InvalidDescriptionHash].upcast[DescriptionHashTag])
```
Comment (sstone) [inline eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala]: A comment that explains where `260` comes from (maybe a quote from BOLT11) may help here

File: eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala [hunk 2]
```diff
@@ -158,11 +158,12 @@ object PaymentRequest {
...
   sealed trait TaggedField
 
-  sealed trait UnknownTaggedField extends TaggedField
+  sealed trait UnknownTaggedField extends TaggedField {
+    def data: BitVector
+  }
 
   // @formatter:off
   case class UnknownTag0(data: BitVector) extends UnknownTaggedField
-  case class UnknownTag1(data: BitVector) extends UnknownTaggedField
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala]: Nice catch

**Comment by pm47** [comment]
Here is my take: https://github.com/ACINQ/eclair/tree/ignore-fields-payment-req

**Comment by t-bast** [comment]
Closing in favor of #1308

**Pull request closed without merging**